### DISCUSSION TITLE
[Feature Request] Explorer APIs

### DISCUSSION THREAD

**Discussion by dev-cz** [zkSync-Community-Hub/zksync-developers#35] [Feature requests]
### Network

Both

### Provide a brief description of the feature you would like to see implemented.

Explorer APIs should be a core feature with any chain. If this is not a high priority it should be, as this is a roadblock for builders.

### Who does this feature request benefit the most (users, developers, protocol, etc.)?

Everyone

### Additional Details

Retrieving Contract Source Code, ABI, internal, external TXs, are all standard features.

_No response_

**Comment by dutterbutter** [comment]
🙌 Thank you for your feature request! We appreciate your contribution to our community. We're keeping track of all the suggestions to prioritize features effectively. An available explorer API for builders is important and will be prioritized accordingly. We'll keep you updated on any progress. 🚀  Please keep the feature requests coming ! @dev-cz

**Comment by keytwojohn** [comment]
Implementing Explorer APIs as a core feature for any blockchain network is essential for fostering developer innovation and user accessibility. By providing easy access to contract source code, ABI, and transaction details, we empower developers to build robust applications while enhancing transparency and usability for all users. Prioritizing this feature is crucial for removing barriers to entry and driving broader adoption across the ecosystem.

**Comment by kimballthomas** [comment]
Implementing Explorer APIs as a core feature for any blockchain network is essential for providing transparency and facilitating development. This feature benefits users, developers, and the protocol by enabling easy access to contract source code, ABI, and transaction details, thereby enhancing usability and fostering innovation.

**Comment by tovanpanet** [comment]
Implementing Explorer APIs as a core feature benefits users, developers, and protocols by providing essential functionality for building, debugging, and interacting with blockchain networks.

**Comment by amandakob** [comment]
Implementing Explorer APIs as a core feature benefits users, developers, and the protocol by providing essential functionality for building and interacting with the blockchain. Retrieving contract details, transaction information, and other data becomes more accessible, facilitating development and usage across the ecosystem.

**Comment by shannonzimmons** [comment]
The user is requesting that Explorer APIs be implemented as a core feature on any chain, as it is currently a roadblock for builders. This feature would benefit everyone, including users, developers, and the protocol. Standard features such as retrieving contract source code, ABI, and internal/external transactions should be included in the Explorer APIs. Further discussion and implementation may be needed to determine the feasibility and impact of this feature on both networks.

**Comment by dutterbutter** [comment]
This is completed.
